apps/ledger-live-desktop/src/newArch/WalletSync/Flows/ManageBackup/useDeleteData.ts

```diff
@@ -0,0 +1,36 @@
+import { useLiveAuthenticate, useTrustchainSdk } from "../../useTrustchainSdk";
+import { useDispatch, useSelector } from "react-redux";
+import {
+  trustchainSelector,
+  memberCredentialsSelector,
+  resetTrustchainStore,
+} from "@ledgerhq/trustchain/store";
+import { useMutation } from "@tanstack/react-query";
+import { JWT } from "@ledgerhq/trustchain/types";
+import { setFlow } from "~/renderer/actions/walletSync";
+import { Flow, Step } from "~/renderer/reducers/walletSync";
+
+export function useDeleteData() {
+  const dispatch = useDispatch();
+  const sdk = useTrustchainSdk();
+  const trustchain = useSelector(trustchainSelector);
+  const memberCredentials = useSelector(memberCredentialsSelector);
+
+  if (!trustchain || !memberCredentials) {
+    throw new Error("trustchain or memberCredentials is missing");
```

Comment:
dangerous to throw in a hook. 
to me, instead we should return a callback that will do nothing if trustchain is falsy, because in that case there will be no more things to delete.